### PR TITLE
not giving up complete disc in case ripping a track fails

### PR DESCRIPTION
**Pull request by fondemen**
Hi there,

It would be nice if ripping could continue even in case ripping a track fails.

Cheers,

### CODE REVIEW

File: whipper/command/cd.py [hunk 1]
```diff
@@ -525,7 +525,12 @@ def ripIfNotRipped(number):
                 track.indexes[1].relative = 0
                 continue
 
-            ripIfNotRipped(i + 1)
+            try:
+                ripIfNotRipped(i + 1)
+            except Exception as e:
```
Comment (RecursiveForest) [inline whipper/command/cd.py]: In general it's considered bad form to catch *all* exception classes with Exception in python, especially if they're not re-raised. (What if a KeyboardInterrupt was generated during ripIfNotRipped?) I know morituri has done this in at least one place, but that doesn't mean we want to introduce more.

Consider figuring out which specific exception you want to catch (perhaps the RuntimeError?), or modify ripIfNotRipped to either raise a semantic exception class or return a value indicating success or failure instead.

File: whipper/command/cd.py [hunk 2]
```diff
@@ -525,7 +525,12 @@ def ripIfNotRipped(number):
...
+                sys.stdout.write(
+                    'ERROR: skipping data track %d, because of: %s\n' % (
```
Comment (RecursiveForest) [inline whipper/command/cd.py]: are you sure it's a *data* track?

**Comment by RecursiveForest** [comment]
I'm not sure if this change, independent of its implementation, is something we want in whipper. whipper is designed to be a tool to produce as close to *perfectly accurate* rips of (currently redbook) CDs. I don't think it should produce incomplete rips; if you have a CD you can't rip all the tracks of for some reason, I would just use cdparanoia directly.

**Comment by fondemen** [comment]
Hello.

I'm not sure you fully got the behavior of this patch. The purpose is to continue ripping a disc when a track is faulty. The faulty track is given up, but next ones remain scheduled. E.g., on a 12 tracks disk with a faulty track 6, whipper rips from track 1 to 5. The patch just makes whipper also rip tracks 7 to 12 (track 6 is given up by the exception).

If you want whipper to rip discs perfectly, and only perfectly, then, in the given case, ripped tracks 1 to 5 should be erased. This would sound weird to me, but this would be consistent with the reasons you invoke for rejecting this patch.

Cheers.

**Comment by MerlijnWajer** [comment]
Reopened.I think you're right in the sense that a best-effort attempt might be nice to have. We should probably discuss what a faulty track entails, and how that would printed in the logs and worked around. In my experience, cdparanoia will either just complete (and you would get a crappy quality track, but this PR would not exist), or cdparanoia will just run pretty much forever. I agree with the general code review that the current PR is no up to the standard we'd expect or want, but I think we should have the discussion.

**Comment by fondemen** [comment]
Great. Thanks for re-opening. I'll update the code to reflect other comments soon.

**Comment by DarkLinkXXXX** [comment]
@fondemen Soon™.
Should I fork and PR myself are or you gonna get back to this?

**Comment by JoeLametta** [comment]
No updates in a year.
Let's keep discussing this feature in #225.

**Pull request closed without merging**